src/models/Ising/IsingModel.jl

```diff
@@ -149,8 +154,10 @@ Calculate energy of Ising configuration `conf` for Ising model `m`.
 """
 function energy(mc::MC, m::IsingModel, conf::IsingConf)
     E = 0.0
-    @inbounds for (source, target) in neighbors(m.l)
-        E -= conf[source] * conf[target]
+    for n in 1:ndims(m)
```

Comment:
This only works for cubic lattices. For example, Honeycomb should have `n = 1:3`, triangular should alternate between `n = 1:1` and `n = 1:2`.

Reply:
If you replace `ndims(m)` by `size(m.neighs, 1)` it should work for arbitrary lattices (which implement `neighbors_lookup_table(l)`).

Reply:
`size(m.neighs, 1)` would count every bond twice. It should be `div(size(m.neighs, 1), 2)`, but that is problematic if `size(m.neighs, 1)` is odd and it relies on `m.neighs` to be sorted in a specific way. (I.e. if `j in m.neighs[i, 1:div(end, 2)], then `i in m.neighs[i, div(end, 2):end]` must be true.)

Reply:
You're right, my bad.